### DISCUSSION TITLE
Database function that is only callable from inside other DB functions

### DISCUSSION THREAD

**Discussion by bombillazo** [supabase/supabase#13411] [General]
Hello, is there a way to use Supabase/Postgres to define a new Database function that can only be called inside other DB functions? Like having it be internal only or something? I don't want this function to be exposed through the API but want to reuse it across different functions.

**Comment by steve-chavez** [comment]
Yes, you can do that like:

```sql
CREATE SCHEMA private;

CREATE FUNCTION private.my_function...
```

All database objects outside of the `public` schema won't be exposed through the API.

**Reply by bombillazo** [reply]
Yup! I was able to do so a while after posting this. Working like a charm.
